frontend/src/components/projectEdit/partnersForm.js

```diff
@@ -0,0 +1,278 @@
+import { useEffect, useState, forwardRef, useMemo } from 'react';
+import { useParams } from 'react-router-dom';
+
+import { useSelector } from 'react-redux';
+import Select from 'react-select';
+import ReactDatePicker from 'react-datepicker';
+import { FormattedMessage } from 'react-intl';
+import { useQuery, useMutation, useQueryClient } from '@tanstack/react-query';
+import { format } from 'date-fns';
+import toast from 'react-hot-toast';
+import PropTypes from 'prop-types';
+
+import messages from './messages';
+import { Alert } from '../alert';
+import { ChevronDownIcon, CloseIcon } from '../svgIcons';
+import { Button } from '../button';
+import { styleClasses } from '../../views/projectEdit';
+import { fetchLocalJSONAPI, pushToLocalJSONAPI } from '../../network/genericJSONRequest';
+import { Listing } from './partnersListing';
+
+export const DateCustomInput = forwardRef(
+  ({ value, onClick, date, handleClear, isStartDate = true, hideCloseIcon = false }, ref) => {
+    return (
+      <div className="relative">
+        <FormattedMessage {...messages[isStartDate ? 'partnerStartDate' : 'partnerEndDate']}>
+          {(message) => {
+            return (
+              <input
+                className="ba b--grey-light pa2 pointer"
+                onClick={onClick}
+                ref={ref}
+                value={value}
+                style={{ height: '42px', boxSizing: 'border-box', maxWidth: '9rem' }}
+                placeholder={message}
+              />
+            );
+          }}
+        </FormattedMessage>
+
+        {(date && hideCloseIcon) || !date ? (
```

Comment:
this can be refactored as `{((date && hideCloseIcon) || !date) && (` and remove the ternary operation.